### DISCUSSION TITLE
How to use relative paths for icon?

### DISCUSSION THREAD

**Discussion by qianxuu** [GitHub30/win11toast#7] [Q&A] [answered]
This is my code and it runs with no response. It works fine with absolute paths.
```python
from win11toast import toast

toast('icon', icon='RazerIcon.ico')
```

**Comment by GitHub30** [comment, marked as answer]
Thank you for your comment. You can make it an absolute path with Pathlib.

```python
from pathlib import Path
from win11toast import toast

toast('icon', icon=str(Path('RazerIcon.ico').resolve()))
```
![image](https://user-images.githubusercontent.com/12811398/197383672-81ddc7f6-1397-4f8c-a639-ee042ca710e5.png)

or use http url
```python
from win11toast import toast

toast(icon='https://raw.githubusercontent.com/iTZUNAMI/ahk-razersynapse3-deamon/master/RazerIcon.ico', app_id='Microsoft.WindowsTerminal_8wekyb3d8bbwe!App')
```
![image](https://user-images.githubusercontent.com/12811398/197383660-64afdf7f-3ef7-4ff6-bc6a-64e69001ca92.png)

**Reply by qianxuu** [reply]
Thanks, does the app_id here apply to Win32 programs?

**Reply by GitHub30** [reply]
I’m not sure of the details.

**Comment by gmandian** [comment]
Hi, I tried an example from the homepage but the icon is not displaying in my notifications. Do you know what the reason might be?
![test](https://user-images.githubusercontent.com/110664746/205997655-d46dca4a-71aa-498a-9b88-e93cbd9f81ad.jpg)

**Reply by GitHub30** [reply]
Thank you for your comment. You can use app_id argument

```python
from win11toast import toast

toast(icon='https://raw.githubusercontent.com/iTZUNAMI/ahk-razersynapse3-deamon/master/RazerIcon.ico', app_id='Microsoft.WindowsTerminal_8wekyb3d8bbwe!App')
```

**Reply by gmandian** [reply]
![toast](https://user-images.githubusercontent.com/110664746/206199293-9ee83f01-e611-42b5-a545-80a05c8ab3d5.jpg)

This is the notification, seems like the app icon is still missing. Could it have something to do with my windows settings? Also, is there a way to change the font size and color of the text? Thanks, this is a really good package besides this one problem ;)

**Reply by GitHub30** [reply]
Thank you for your comment. Please tell me the result of executing `Get-StartApps`.
```powershell
Get-StartApps
```
![image](https://user-images.githubusercontent.com/12811398/206333422-9d108bc2-1789-428c-ad8d-3bfbe6251deb.png)

```python
toast(icon='https://raw.githubusercontent.com/iTZUNAMI/ahk-razersynapse3-deamon/master/RazerIcon.ico', app_id='Microsoft.WindowsCalculator_8wekyb3d8bbwe!App')
```

![image](https://user-images.githubusercontent.com/12811398/206333488-42194ee1-277c-4d6e-b780-99db32a16d52.png)

![image](https://user-images.githubusercontent.com/12811398/206334004-76c0efa3-5375-4c34-9984-d6cf2fcce8a7.png)

For example, if the app_id contains c:internet, the icon can be displayed.

**Reply by gmandian** [reply]
Hi, this solution worked and the logo is displaying. 

I want to ask about button click events, can I specify a function to be called directly in the arguments field of the button objects? So when I click on a button in the notification, it performs a certain task? Something like this:

buttons = [
    {'arguments': performtaskA, 'content': 'A'},
    {'arguments': performtaskB, 'content': 'B'}
]

toast('title','text',buttons=buttons)

**Reply by GitHub30** [reply]
Thank you for your comment. You can use `on_click` argument.

```python
from win11toast import toast

def performtaskA():
    print('performtaskA called')

def performtaskB():
    print('performtaskB called')

buttons = [
{'arguments': 'performtaskA', 'content': 'A'},
{'arguments': 'performtaskB', 'content': 'B'}
]

toast('title','text',buttons=buttons, on_click=lambda arg: eval(arg['arguments'])())
```


https://user-images.githubusercontent.com/12811398/206859753-fba14019-55c2-40b3-a0fe-28d8d43cb11d.mp4

**Reply by gmandian** [reply]
Hi, thanks for your help again. I'm just coming back to this today 

```
# functions above

toast('title', 'text', buttons=buttons, icon=icon, app_id='Microsoft.WindowsCalculator_8wekyb3d8bbwe!App', on_click=on_click)
```

I have the notifications, logo and buttons are all working just fine but it launches the calculator app when clicked, is there a way to prevent that? 

I tried adding `'activationType': 'background'` in `buttons` but then it won't run my code.

**Comment by SurajBhari** [comment]
I am trying to achieve that a notification is sent on media change from the app_id of the source application too 
I have a session obj [like this](https://learn.microsoft.com/en-us/uwp/api/windows.media.control.globalsystemmediatransportcontrolssession?view=winrt-22621) and using [this property](https://learn.microsoft.com/en-us/uwp/api/windows.media.control.globalsystemmediatransportcontrolssession.sourceappusermodelid?view=winrt-22621#windows-media-control-globalsystemmediatransportcontrolssession-sourceappusermodelid)
I can somewhat achieve this 

the property returns `Edge` or `Chrome` The issue is it works only on those long looking app id and not the short one like the terminal one 
![image](https://github.com/GitHub30/win11toast/assets/45149585/0eb013c8-cc05-425a-8409-6482aa9337f7)

**Reply by SurajBhari** [reply]
Get-StartApps 
returns 
![image](https://github.com/GitHub30/win11toast/assets/45149585/fb7c095d-78d4-4c95-abc0-729c59dfc8cf)
which doesn't show the image

**Reply by SurajBhari** [reply]
fixed the linking.
